### PR TITLE
Clean up external tables related code

### PR DESCRIPTION
**Pull request by hlinnaka**
To make it nicer to work with. There are no functional changes here.

### CODE REVIEW

**Comment by danielgustafsson** [comment]
The curl implementation sprinkles asserts like `Assert (file->type == CFTYPE_CURL);` while the others don't. Is there value in being consistent and do that for the other URL types as well? Not that it would be easy to trip those asserts..

File: gpAux/extensions/gphdfs/gphdfsprotocol.c [hunk 1]
```diff
@@ -401,7 +402,9 @@ gphdfsprotocol_import(PG_FUNCTION_ARGS)
 	datlen 	= EXTPROTOCOL_GET_DATALEN(fcinfo);
 
 	if (datlen > 0 && !myData->importDone)
-		nread = piperead(myData->importFile->u.exec.pipes[EXEC_DATA_P], data, datlen);
+	{
+		nread = url_execute_fread(data, datlen, myData->importFile, NULL);
+	}
```
Comment (danielgustafsson) [inline gpAux/extensions/gphdfs/gphdfsprotocol.c]: Why the added parenthesis?

File: src/backend/access/external/fileam.c [hunk 2]
```diff
@@ -2041,11 +2050,11 @@ strtokx2(const char *s,
 
 	if (s)
 	{
-		//pfree(storage);
+		/* pfree(storage); */
```
Comment (danielgustafsson) [inline src/backend/access/external/fileam.c]: Should we just kill this line?

**Comment by danielgustafsson** [comment]
It's admittedly easy to miss things in large diffs, but nothing really sticks out and it looks good. :+1:

File: gpAux/extensions/gphdfs/gphdfsprotocol.c (hunk 1 above)
Comment (hlinnaka) [inline gpAux/extensions/gphdfs/gphdfsprotocol.c]: No reason. Removed.

File: src/backend/access/external/fileam.c (hunk 2 above)
Comment (hlinnaka) [inline src/backend/access/external/fileam.c]: Yes. Killed.

**Comment by hlinnaka** [comment]
Pushed, after fixing those things you pointed out. I removed the Asserts that you mentioned. They might be marginally useful safety checks, if we had them in all the implementations, but meh.

Thanks for the review!

**Pull request closed without merging**